src/classes/frDonation.cls

```diff
@@ -53,84 +55,126 @@ public class frDonation extends frModel implements frSyncable {
         return mappings;
     }
     
-    private Opportunity o;
+    private SObject o;
+    private SObject gt;
     
     public frDonation(Sync_Attempt__c syncRecord){
         super(syncRecord);
     }
-    
+
     public Boolean sync() {
+        //create Sobject
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        o = new Opportunity();
+        String frId = getFunraiseId();
+
+        if(!frUtil.hasNPCobjects()) {
+            o = new Opportunity();
+            applyMappings(o, request);
+            setOpportunityStage((Opportunity) o, String.valueOf(request.get('status')));
+        } 
+        else {
+            List<SObject> gtList = Database.query('SELECT Id, Name, DonorId, AcknowledgementDate FROM GiftTransaction WHERE fr_ID__c = :frId');
+            if(gtList.isEmpty()) {
+                o = Schema.getGlobalDescribe().get('gifttransaction').newSObject();
+            } 
+            else {
+                o = gtList.get(0);
+            }
+            o.put('Name', String.valueOf(request.get('name')));
+            o.put('OriginalAmount', (Decimal) request.get('amount'));
+            o.put('PaymentMethod', String.valueOf(request.get('paymentMethodType')));
+            o.put('AcknowledgementDate', System.today());
+        }
+
         applyMappings(o, request);
-        setOpportunityStage(o, String.valueOf(request.get('status')));
-        
         String funraiseId = getFunraiseId();       
-        String supporterFunraiseId = String.valueOf(request.get('donorId'));        
+        String supporterFunraiseId = String.valueOf(request.get('donorId')); 
+        String supporterObjName = frUtil.hasPersonAccountsEnabled() ? 'Account' : 'Contact';
+        String subsciptionObjName = frUtil.hasNPCobjects() ? 'GiftCommitment' : 'Subscription__c';
+        
+        //connect supporter
         if (String.isNotBlank(supporterFunraiseId)) {
-            List<Contact> contacts = [SELECT Id, AccountId FROM Contact WHERE fr_ID__c = :supporterFunraiseId];
-            if (contacts.size() > 0) {
-                Contact supporterContact = contacts.get(0);
-                o.fr_Donor__c = supporterContact.Id;
-                if(frUtil.hasNPSP()){
-                    o.put('npsp__Primary_Contact__c', supporterContact.Id);
```

Comment:
It looks like this code has been removed,  but it shouldn't have been.  We should still be supporting if the NPSP is installed and populating this lookup field

Reply:
Sorry missed this! Added back in new update, if im missing any other existing functionality lmk and ill get right on it ty!